src/rust/engine/process_executor/src/remote.rs

```diff
@@ -1,23 +1,57 @@
 use std::error::Error;
+use std::sync::Arc;
 
 use bazel_protos;
 use crypto::digest::Digest;
 use crypto::sha2::Sha256;
+use grpcio::{ChannelBuilder, Environment};
 use protobuf;
+use protobuf::Message;
 
 use super::{ExecuteProcessRequest, ExecuteProcessResult};
 
 ///
 /// Runs a command via a gRPC service implementing the Bazel Remote Execution API
 /// (https://docs.google.com/document/d/1AaGk7fOPByEvpAbqeXIyE8HX_A3_axxNnvroblTZ_6s/edit).
 ///
-pub fn run_command_remote(req: ExecuteProcessRequest) -> Result<ExecuteProcessResult, String> {
+pub fn run_command_remote(addr: &str, req: ExecuteProcessRequest) -> Result<ExecuteProcessResult, String> {
+  let execute_request = make_execute_request(&req)?;
+
+  let env = Arc::new(Environment::new(1));
+  let channel = || {
+    ChannelBuilder::new(Arc::clone(&env)).connect(addr)
```

Comment:
Should be able to `env.clone()` rather than calling `Arc::clone(&env)`.

Reply:
Done